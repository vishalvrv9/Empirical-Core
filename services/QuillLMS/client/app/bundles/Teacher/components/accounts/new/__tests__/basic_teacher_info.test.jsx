import React from 'react';
import { shallow } from 'enzyme';

import BasicTeacherInfo from '../basic_teacher_info';

describe('BasicTeacherInfo component', () => {

  it('renders a signup form', () => {
    const wrapper = shallow(
        <BasicTeacherInfo signUp={() => null} update={() => null} errors={{}}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('applies textInputGenerator generate function to form fields', () => {

    const formFields = [
        {
            name: 'first_name',
            label: 'First Name',
            errorKey: 'name',
            errorLabel: 'Name'
        }, {
            name: 'last_name',
            label: 'Last Name',
            errorKey: 'name',
            errorLabel: 'Name'
        }, {
            name: 'email'
        }, {
            name: 'password'
        }
    ];

    const mockGenerator = jest.fn();
    const wrapper = shallow(
        <BasicTeacherInfo signUp={() => null} update={() => null}  errors={{}} />
    );
  });

});
