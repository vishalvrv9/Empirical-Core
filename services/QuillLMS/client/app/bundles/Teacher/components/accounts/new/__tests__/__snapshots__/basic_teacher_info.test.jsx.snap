// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTeacherInfo component renders a signup form 1`] = `
<div
  className="new-teacher-account"
>
  <h3
    className="sign-up-header"
  >
    Sign up for a Teacher Account
  </h3>
  <p
    className="support-p text-center"
  >
    We now support Google Classroom!
  </p>
  <auth_sign_up />
  <div
    className="need-a-border"
  />
  <div
    className="text-input-row"
  >
    <div
      className="form-label"
    >
      First Name
    </div>
    <input
      id="first_name"
      onChange={[Function]}
      placeholder="First Name"
      type="text"
    />
    <span />
  </div>
  <div
    className="text-input-row"
  >
    <div
      className="form-label"
    >
      Last Name
    </div>
    <input
      id="last_name"
      onChange={[Function]}
      placeholder="Last Name"
      type="text"
    />
    <span />
  </div>
  <div
    className="text-input-row"
  >
    <div
      className="form-label"
    >
      Email
    </div>
    <input
      id="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
    />
    <span />
  </div>
  <div
    className="text-input-row"
  >
    <div
      className="form-label"
    >
      Password
    </div>
    <input
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <span />
  </div>
  <input
    name="sendNewsletter"
    onChange={[Function]}
    type="checkbox"
  />
  <p>
    Send me monthly Quill updates
  </p>
  <button
    className="sign-up-button button-green"
    onClick={[Function]}
  >
    Sign Up
  </button>
  <div
    className="text-align-center"
  >
    By signing up, you agree to our 
    <a
      href="/tos"
      target="_blank"
    >
      terms of service
    </a>
     and 
    <a
      href="/privacy"
      target="_blank"
    >
      privacy policy
    </a>
    .
  </div>
</div>
`;
